core/node/base_token_adjuster/src/metrics.rs

```diff
@@ -0,0 +1,34 @@
+use std::time::Duration;
+
+use vise::{Buckets, EncodeLabelSet, EncodeLabelValue, Family, Histogram, Metrics};
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, EncodeLabelSet, EncodeLabelValue)]
+#[metrics(label = "operation_result", rename_all = "snake_case")]
+pub(super) enum OperationResult {
+    Success,
+    Failure,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, EncodeLabelSet)]
+pub(crate) struct OperationResultLabels {
+    pub result: OperationResult,
+    pub attempts: u32,
+}
+
+/// Roughly exponential buckets for gas (10k – 50M).
+const GAS_BUCKETS: Buckets =
+    Buckets::values(&[1e4, 2e4, 5e4, 1e5, 2e5, 5e5, 1e6, 2e6, 5e6, 1e7, 2e7, 5e7]);
+
+#[derive(Debug, Metrics)]
+#[metrics(prefix = "snapshots_creator")]
+pub(crate) struct BaseTokenAdjusterMetrics {
+    #[metrics(buckets = GAS_BUCKETS)]
+    pub l1_gas_used: Histogram<f64>,
```

Comment:
how often is this reported?  Do we update L1 values more often than once per minute or similar? Because if not, I'd just use a gauge and see the exact gas value (as opposed to the buckets)

Reply:
Agreed - switching to gauge